ext/gd/libgd/gd_avif.c

```diff
@@ -342,6 +342,14 @@ gdImagePtr gdImageCreateFromAvifCtx (gdIOCtx *ctx)
 
 	decoder = avifDecoderCreate();
 
+	// Check if libavif version is >= 0.9.1
+#if AVIF_VERSION_MAJOR > 0 || (AVIF_VERSION_MAJOR == 0 && AVIF_VERSION_MINOR > 9) || (AVIF_VERSION_MAJOR == 0 && AVIF_VERSION_MINOR == 9 && AVIF_VERSION_PATCH >= 1)
```

Comment:
Is this version based conditional compilation necessary? Can't we just always exclude this flag? If `AVIF_STRICT_PIXI_REQUIRED` is only available as of 0.9.2, we could check whether it is defined instead of checking the version.

Reply:
Hi Christoph,

AVIF_STRICT_PIXI_REQUIRED is defined as an enumerator, not as a macro. So we can't say

`#if defined(AVIF_STRICT_PIXI_REQUIRED)`

The version check could be simplified if necessary by packing major, minor, and patch into an integer and compare the integers. For example,

```
#define AVIF_VERSION_0_9_1 ((0 * 1000000) + (9 * 10000) + (1 * 100) + 0)
#if AVIF_VERSION >= AVIF_VERSION_0_9_1
```

This assumes the way `AVIF_VERSION` is defined in <avif/avif.h> will never change:

```
#define AVIF_VERSION \
    ((AVIF_VERSION_MAJOR * 1000000) + (AVIF_VERSION_MINOR * 10000) + (AVIF_VERSION_PATCH * 100) + AVIF_VERSION_DEVEL)
```

Reply:
> AVIF_STRICT_PIXI_REQUIRED is defined as an enumerator, not as a macro.

Ah, then the version check is fine for me.